src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -1417,6 +1418,66 @@ public void TestProjectCompletenessWithMultipleProjects()
             Assert.True(transitivelyDependsOnNormalProjects.HasSuccessfullyLoadedAsync().Result);
         }
 
+        [Fact]
+        public async Task TestMassiveFileSize()
+        {
+            // set max file length to 1 bytes
+            var maxLength = 1;
+            var workspace = new AdhocWorkspace(TestHost.Services, ServiceLayer.Host);
+            workspace.Options = workspace.Options.WithChangedOption(FileTextLoaderOptions.FileLengthThreshold, maxLength);
+
+            string tempPath = null;
+            try
+            {
+                // save temp file
+                tempPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
```

Comment:
sure, will change to use that. I searched how other does this kind of thing in code base and did same thing.